app/lib/features/home/widgets/my_spaces_section.dart

```diff
@@ -18,68 +20,122 @@ class MySpacesSection extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
+    final bookmarkedSpaces = ref.watch(bookmarkedSpacesProvider);
     final spaces = ref.watch(spacesProvider);
+    if (bookmarkedSpaces.isNotEmpty) {
+      return _RenderSpacesSection(
+        spaces: bookmarkedSpaces,
+        limit: bookmarkedSpaces.length,
+        showAll: true,
+        showAllCounter: spaces.length,
+        title: Row(
+          children: [
+            const Icon(Icons.bookmark_outline),
+            Text(
+              L10n.of(context).bookmarkedSpaces,
+              style: Theme.of(context).textTheme.titleMedium,
+            ),
+          ],
+        ),
+      );
+    }
+
+    // fallback
+    if (spaces.isEmpty) {
+      return const _NoSpacesWidget();
+    }
 
     int spacesLimit =
         (limit != null && spaces.length > limit!) ? limit! : spaces.length;
+    return _RenderSpacesSection(
+      spaces: spaces,
+      limit: spacesLimit,
+      showAll: spacesLimit != spaces.length,
+      showActions: spacesLimit == spaces.length,
+      showAllCounter: spaces.length,
+      title: InkWell(
+        key: DashboardKeys.widgetMySpacesHeader,
+        onTap: () => context.pushNamed(Routes.spaces.name),
+        child: Text(
+          L10n.of(context).mySpaces,
+          style: Theme.of(context).textTheme.titleMedium,
+        ),
+      ),
+    );
+  }
+}
+
+class _RenderSpacesSection extends ConsumerWidget {
+  final int limit;
+  final List<Space> spaces;
+  final Widget title;
+  final bool showAll;
+  final bool showActions;
+  final int showAllCounter;
```

Comment:
Instead of `showAll` and `showActions` param, we can use param like `bookmarkOnly` bool param. 

And also `showAllCounter` may not required because we can get it from spaces list directly which we are already passing in param.

Reply:
This isn't about the bookmarked ones but also if we have a longer total list than the one we are currently showing. I'd rather keep them explicit in their nature rather than imply that from context.